packages/apputils-extension/src/statusbarplugin.ts

```diff
@@ -188,25 +195,66 @@ export const runningSessionsStatus: JupyterFrontEndPlugin<void> = {
     if (settingRegistry) {
       let disposable: IDisposable;
       const onSettingsUpdated = (
-        settings: ISettingRegistry.ISettings
+        kernelSettings?: ISettingRegistry.ISettings,
+        terminalsSettings?: ISettingRegistry.ISettings
       ): void => {
-        const showStatusBarItem = settings.get('showStatusBarItem')
-          .composite as boolean;
+        const showTerminalsMap = {
+          'if-any': undefined,
```

Comment:
If the `showTerminals` setting is set to 'if-any' and the `showKernels` set to `false`, the `RunningSession` item is disposed.

 [record-2025-03-21_12.33.30.webm](https://github.com/user-attachments/assets/f17a4b46-5de3-4f69-9635-68081c31a8e8)

Reply:
Thank you, 08d2daa should fix it